packages/schema/src/config/vite.ts

```diff
@@ -30,6 +30,7 @@ export default defineUntypedSchema({
       extensions: ['.mjs', '.js', '.ts', '.jsx', '.tsx', '.json', '.vue']
     },
     publicDir: {
+      $schema: { deprecated: 'use `dir.public` option instead' },
```

Comment:
Just to confirm, is this used for documentation auto-generation and no influence of coding actual application code?

Reply:
By the way, the message is quoted from `dir.static` option.

Reply:
Although I deprecate it at docs level, developers still can specify `vite.publicDir` option in Nuxt config file, because Nuxt accepts `ViteConfig` as `vite` option that  extends/inherits all the properties from Vite's `UserConfig`.

Reply:
yes, that's tricky. Setting `$schema` here won't have any effect, as you say, because it's being shadowed by `ViteConfig`. So instead we can set the deprecation on `ViteConfig` directly.